### DISCUSSION TITLE
How to combine Textinput and Dropdown?

### DISCUSSION THREAD

**Discussion by balabas** [rio-labs/rio#118] [Q&A] [answered]
I need to input text with autocomplete, so I can create copy of Dropdown class with modification that catch "_on_message" event, but "Dropdown-builtin" class in fronted assets does not send messages on keypress. So, I need can't figure out, how to override it without changing package. Is it possible?

**Comment by mad-moo** [comment]
Hey! I think your best bet might be creating your own component. Have you tried adding a `rio.Popup` and putting your autocomplete options there?

**Reply by balabas** [reply]
Thanks for advice, but how i can create own component with custom frontend part? I can not find any instructions in documentation.
Popup could be solution if I could catch arrow keys from Textinput.

**Reply by mad-moo** [reply, marked as answer]
Sorry, that's not what I meant. I don't think you need to resort to custom HTML. Just create a python class, and inherit from `rio.Component`, like all components do.

In that class, combine a `rio.TextInput` with a `rio.Popup`. I think that should work for your use case
